### PR TITLE
Do sky subtraction on original images, use 

### LINKED ISSUE #116: do sky subtraction on the coadd not the sheared images

### PR DESCRIPTION
**Pull request by esheldon**
closes #116 

this will fail until this PR is merged:  https://github.com/LSSTDESC/descwl-shear-sims/pull/157  because it is needed for the tests

### CODE REVIEW

**Comment by esheldon** [comment]
also actual tests recovering shear with a non zero sky, and with EDGE making it into the coadd, will fail until this one is merged: https://github.com/LSSTDESC/descwl_coadd/pull/62

**Review by beckermr — CHANGES_REQUESTED** [review]
Can you provide a bit more context for this PR?

File: metadetect/lsst_measure.py [hunk 1]
```diff
@@ -284,6 +276,32 @@ def iterate_detection_and_skysub(
     return result
 
 
+def subtract_sky_mbobs(mbobs, thresh):
+    """
+    subtract sky and zero the edges
```
[the remark this answers is not in the document]
Comment (esheldon) [reply metadetect/lsst_measure.py:281]: will remove

**Comment by esheldon** [comment]
Context

We are running sky subtraction separately for each of the metacal images, but I think we should run it on the input coadd images, then run metacal steps on those

**Comment by esheldon** [comment]
The other part is to use the new functionality from descwl-shear-sims to generate non-zero skys for testing this feature.

**Comment by esheldon** [comment]
Do you need more time on this one @beckermr ?

File: metadetect/tests/test_lsst_metadetect.py [hunk 2]
```diff
@@ -143,8 +143,8 @@ def test_lsst_metadetect_mfrac_ormask(cls):
         for shear in ["noshear", "1p", "1m", "2p", "2m"]:
             assert np.any(res[shear]["flags"] == 0)
             assert np.all(
-                (res[shear]["mfrac"] > 0.45)
-                & (res[shear]["mfrac"] < 0.55)
+                (res[shear]["mfrac"] > 0.40)
+                & (res[shear]["mfrac"] < 0.60)
```
Comment (beckermr) [inline metadetect/tests/test_lsst_metadetect.py:147]: This change in mfrac seems unrelated to this PR. What happened here?

Comment (esheldon) [reply metadetect/tests/test_lsst_metadetect.py:147]: I interpreted it as related to some noise difference.   The same seed may have resulted in slightly different measurements due to small changes in code order

Comment (esheldon) [reply metadetect/tests/test_lsst_metadetect.py:147]: What do you think, is that convincing?

Comment (esheldon) [reply metadetect/tests/test_lsst_metadetect.py:147]: I noticed that failed in your PR as well, so may be related to the stack version

Comment (beckermr) [reply metadetect/tests/test_lsst_metadetect.py:147]: It just seems rather odd. :/

Comment (esheldon) [reply metadetect/tests/test_lsst_metadetect.py:147]: I agree.  Since it affects different unrelated PRs it seems unrelated to this repo at all

Comment (beckermr) [reply metadetect/tests/test_lsst_metadetect.py:147]: Let's not delay and just go for it. If it changes again we can dig.

**Review by beckermr — APPROVED** [review]
